src/app/modules/submission/submission.service.ts

```diff
@@ -251,17 +258,64 @@ export const copyPendingSubmissionToSubmissions = (
       return okAsync(submission)
     })
     .andThen((pendingSubmission) => {
-      const submission = new SubmissionModel(
-        omit(pendingSubmission, ['_id', 'created', 'lastModified']),
-      )
+      const submissionContent = omit(pendingSubmission, [
+        '_id',
+        'created',
+        'lastModified',
+      ])
+      const submission = new SubmissionModel({
+        // Explicitly copy over the pending submission's _id
+        _id: pendingSubmissionId,
+        ...submissionContent,
+      })
 
       return ResultAsync.fromPromise(submission.save({ session }), (error) => {
         logger.error({
           message: 'Database save submission error',
           meta: logMeta,
           error,
         })
-        return new DatabaseError(getMongoErrorMessage(error))
+
+        const databaseError = transformMongoError(error)
+
+        return databaseError instanceof DatabaseDuplicateKeyError
+          ? databaseError
+          : new DatabaseError(getMongoErrorMessage(error))
+      }).orElse((error) => {
+        if (error instanceof DatabaseDuplicateKeyError) {
+          // Failed to save due to duplicate keys.
+          logger.error({
+            message:
+              'Failed to move pending submission to submission: duplicate key error in submission collection',
+            meta: logMeta,
+            error,
+          })
+          // Recover by attempting to save with a different id.
+          const recoverySubmission = new SubmissionModel(submissionContent)
+          // TODO: Set alarms for both branches
+          return ResultAsync.fromPromise(
+            recoverySubmission.save({ session }),
+            (error) => {
+              logger.error({
+                message: 'Failed to recover from duplicate key error',
```

Comment:
Not related to PR but I'm wondering if critical errors like this should have its own error code like `internalCode: P001` instead of a message.

Reply:
it could make our alarm system more fine-grained, if we can set alarms by the error code returned rather than just the status code... maybe one for the tech backlog!